packages/components/resource-pages/list-page.tsx

```diff
@@ -0,0 +1,288 @@
+      }
+    ],
+    [t]
+  );
+
+  const [columns] = useActiveColumns({
+    columns: tableColumns,
+    showNamespaceOverride: false,
+    columnManagementID: null
+  });
+  return (
+    <VirtualizedTable
+      {...props}
+      aria-label={t("Resource Page")}
+      columns={columns}
+      Row={RowRenderer}
+    />
+  );
+};
+
+type CustomData = {
+  launchModal: LaunchModal;
+  resourceModel: K8sModel;
+  kebabActions?: (t: TFunction) => { [key: string]: { value: string } };
+};
+
+const RowRenderer: React.FC<RowProps<K8sResourceCommon, CustomData>> = ({
+  obj,
+  activeColumnIDs,
+  rowData
+}) => {
+  const { resourceModel, launchModal, kebabActions } = rowData;
+  const resource = obj as K8sResourceCommon & {
+    provider?: string;
+    buckets?: number;
+  };
+  return (
+    <>
+      <TableData {...tableColumnInfo[0]} activeColumnIDs={activeColumnIDs}>
+        {getName(obj)}
+      </TableData>
+      <TableData {...tableColumnInfo[1]} activeColumnIDs={activeColumnIDs}>
+        <OperandStatus operand={obj} />
+      </TableData>
+      <TableData {...tableColumnInfo[2]} activeColumnIDs={activeColumnIDs}>
+        {resource.provider}
+      </TableData>
+      <TableData {...tableColumnInfo[3]} activeColumnIDs={activeColumnIDs}>
+        {String(resource?.buckets) + " Buckets"}
+      </TableData>
+      <TableData {...tableColumnInfo[4]} activeColumnIDs={activeColumnIDs}>
+        <Kebab
+          launchModal={launchModal}
+          extraProps={{ resource: obj, resourceModel: resourceModel }}
+          customKebabItems={kebabActions}
+        />
+      </TableData>
+    </>
+  );
+};
+
+type GenericListPageProps = {
+  actions?: ModalMap;
+  resourceModel: K8sModel;
+  kebabActions?: CustomData["kebabActions"];
+};
+
+const DataResourceListView: React.FC<GenericListPageProps> = ({
+  actions,
+  resourceModel,
+  kebabActions
+}) => {
+  const { t } = useTranslation();
+  const [Modal, modalProps, launchModal] = useModalLauncher(actions);
+
+  const bucketResource = React.useMemo(
+    () => ({
+      kind: referenceForModel(NooBaaBucketClassModel),
+      namespace: DATA_FEDERATION_NAMESPACE,
+      isList: true
+    }),
+    []
+  );
+  const [bucketResources] =
+    useK8sWatchResource<BucketClassKind[]>(bucketResource);
+  let bucketClassDict = {};
+  bucketResources.forEach((bc) => {
+    if (bc.spec?.namespacePolicy?.type === BucketClassType.SINGLE) {
+      const ns = bc.spec.namespacePolicy.single.resource;
+      bucketClassDict[ns] = bucketClassDict[ns] + 1 || 1;
+    } else if (bc.spec?.namespacePolicy?.type === BucketClassType.MULTI) {
+      const dataResourceSet = new Set<string>();
+      dataResourceSet.add(bc.spec.namespacePolicy.multi.writeResource);
+      bc.spec.namespacePolicy.multi.readResources.forEach((r) =>
+        dataResourceSet.add(r)
+      );
+      dataResourceSet.forEach(
+        (ns) => (bucketClassDict[ns] = bucketClassDict[ns] + 1 || 1)
+      );
+    }
+  });
```

Comment:
either use term dataresource every where or namespace store everywhere (for variables)...
```suggestion
  bucketClasses?.reduce((bc) => {
  const namespacePolicy = bc?.spec?.namespacePolicy;
    if (namespacePolicy?.type === BucketClassType.SINGLE) {
      const ns = namespacePolicy?.single?.resource;
      bucketClassDict[ns] = bucketClassDict[ns] + 1 || 1;
    } else if (namespacePolicy?.type === BucketClassType.MULTI) {
      const dataResourceSet = new Set<string>();
      dataResourceSet.add(namespacePolicy?.multi?.writeResource);
      namespacePolicy?.multi?.readResources.forEach((r) =>
        dataResourceSet.add(r)
      );
      dataResourceSet.forEach(
        (ns) => (bucketClassDict[ns] = bucketClassDict[ns] + 1 || 1)
      );
    }
  });
```

Reply:
use reduce and chnage the format accordingly... I just changed `forEach` to `reduce` above but above suggestion will not work if you copy paste it directly...